src/pages/settings/tabs/ApiKeys.tsx

```diff
@@ -0,0 +1,148 @@
+import Input from '@/components/Input/Input';
+import toast from '@/components/Toast';
+import { useCreateApiToken, useDeleteApiToken } from '@/core/react-query/auth/mutations';
+import { useApiKeyQuery } from '@/core/react-query/auth/queries';
+import { invalidateQueries } from '@/core/react-query/queryClient';
+import useEventCallback from '@/hooks/useEventCallback';
+
+import type { AuthToken } from '@/core/types/api/authToken';
+
+const UserApiTokens = (prop: { userInfo: AuthToken }) => {
+  const { userInfo: { Device, Username } } = prop;
+  const { mutate: deleteToken } = useDeleteApiToken();
+
+  const onDeleteClick = useEventCallback(() => {
+    deleteToken(Device, {
+      onSuccess: (() => {
+        toast.success('API Key Deleted', `API Key ${Device} has been deleted!`, {
+          autoClose: 3000,
+        });
+        invalidateQueries(['auth', 'apikey']);
+      }),
+    });
+  });
+
+  return (
+    <div className="flex flex-row justify-between">
+      <div className="flex w-[65%] flex-col gap-y-1">
+        <span className="font-light">{Username}</span>
+        <span className="font-normal">{Device}</span>
+      </div>
+      <Button buttonType="danger" className="px-4 py-2" onClick={onDeleteClick}>
+        Delete API Key
+      </Button>
+    </div>
+  );
+};
+
+const ApiKeys = () => {
+  const [keyValue, setKeyValue] = useState('');
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [, copy] = useCopyToClipboard();
+  const onKeyChange = useEventCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    setKeyValue(event.target.value);
+  });
+
+  const {
+    data: createdToken,
+    isError: generatedFailed,
+    isSuccess: generatedSucceed,
+    mutate: createApiToken,
+  } = useCreateApiToken();
+
+  const handleCopyToClipboard = useEventCallback(() => {
+    if (!generatedSucceed) return;
+    copy(keyValue).then((isCopied) => {
+      if (!isCopied) return;
+      toast.success('Copied', 'API Key has been copied to clipboard!', {
+        autoClose: 3000,
+        toastId: 'api-copied',
+      });
+    }).catch(console.error);
+  });
+
+  const onGenerateClick = useEventCallback(() => {
+    createApiToken(keyValue, {
+      onSuccess: () => {
+        setInputDisabled(_ => true);
+        toast.success('API Generated', 'API Key has been generated!', {
+          autoClose: 3000,
+          toastId: 'api-generated',
+        });
+        toast.warning('Copy Your API Key!', 'You won\'t be able to copy this key anymore once you leave this page!', {
+          autoClose: 99999999999,
+          toastId: 'copy-api-key',
+        });
+        invalidateQueries(['auth', 'apikey']);
+      },
+    });
+  });
+
+  const { data: tokens, refetch } = useApiKeyQuery();
+
+  useEffect(() => {
+    if (!generatedSucceed || !createdToken) return undefined;
+    setKeyValue(createdToken);
+    setInputDisabled(true);
+    return () => {
+      if (!toast.isActive('copy-api-key')) return;
```

Comment:
This will bail and potentially leave some toasts open, condition needs to be reversed and return is not needed. Should be:
```
if (toast.isActive('copy-api-key')) {
    toast.dismiss('copy-api-key');
}
```